### PR TITLE
HW 04: by Vadim Volovenko

### PR DESCRIPTION
**Pull request by Volovenko**

### CODE REVIEW

**Comment by vladhilko** [comment]
Please, run `rubocop-rspec` 😔

**Comment by Volovenko** [comment]
> Please, run `rubocop-rspec` 😔

did it

**Comment by vladhilko** [comment]
Can you remove .DS_Store from you pr? Or add this file to the `.gitignore`?

File: HW04/Vadim_Volovenko/implementations_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+      it 'array nill' do
+        expect(subject.valid?).to eq false
+      end
+    end
+  end
+
+  describe '#my_map' do
+    context 'when array valid' do
+      it 'return right array' do
```
Comment (vladhilko) [inline HW04/Vadim_Volovenko/implementations_spec.rb:26]: Your spec description should look like documentation. Be explicit describing your spec, what is a valid array?  What is the right array? 

should be something like this 
```suggestion
    context 'when array is present' do
      it 'transforms an array based on block conditions' do
```

**Comment by Volovenko** [comment]
> Can you remove .DS_Store from you pr? Or add this file to the `.gitignore`?

ok, i deleted that file

File: HW04/Vadim_Volovenko/implementations_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    end
+  end
+
+  describe '#my_each' do
+    context 'when strings valid' do
+      it 'return right string' do
+        expectation = expect do
+          (subject.my_each { |element| puts element * 2 })
+        end
+        expectation.to output("2\n4\n6\n").to_stdout
```
Comment (vladhilko) [inline HW04/Vadim_Volovenko/implementations_spec.rb:46]: we don't use it in this way 
```suggestion
        expect do
          (subject.my_each { |element| puts element * 2 })
        end.to output("2\n4\n6\n").to_stdout
```

**Comment by vladhilko** [comment]
> > Can you remove .DS_Store from you pr? Or add this file to the `.gitignore`?
> 
> ok, i deleted that file

not everywhere, check file changes

**Comment by Volovenko** [comment]
> > > Can you remove .DS_Store from you pr? Or add this file to the `.gitignore`?
> > 
> > 
> > ok, i deleted that file
> 
> not everywhere, check file changes

done

**Pull request merged**